api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -439,6 +445,35 @@ private void attemptFirecloudWorkspaceCreation(FirecloudWorkspaceId workspaceId)
     return ResponseEntity.ok(new EmptyResponse());
   }
 
+  @Override
+  public ResponseEntity<List<FileDetail>> getNoteBookList(String workspaceNamespace, String workspaceId) {
+    List<FileDetail> bucketFileList = new ArrayList<FileDetail>();
+    try {
+      org.pmiops.workbench.firecloud.model.WorkspaceResponse workspaceResponse = fireCloudService.getWorkspace(workspaceNamespace, workspaceId);
+      if(workspaceResponse!=null) {
+        org.pmiops.workbench.firecloud.model.Workspace fireCloudWorkspace = workspaceResponse.getWorkspace();
+        if (fireCloudWorkspace != null) {
+          String bucketName = fireCloudWorkspace.getBucketName();
+          bucketFileList = cloudStorageService.getBucketFileList(bucketName);
```

Comment:
+1. 

One proposal for how this all works.

- We have a /notebooks dir which contains notebook files. We may also have an /input dir that contains other data files (provided by researchers) which the notebooks need to run. And we have the all_of_us_config.json file created when a workspace is created.
- When somebody launches a cluster, we automatically copy all of these files over. (The user doesn't have to choose which files to copy.)
- Ideally, when the user saves the notebook in the cluster, it gets written back to the GCS bucket automatically. (Some question around the UX for locking here which we're still discussing.)

Reply:
I think we should have a break out meeting on this. We definitely don't want to copy all files over, because then it can get messy and have people clobbering other people's work in collaborative workspaces where multiple people are running clusters.

Reply:
Yeah, that's where the whole "locking" thing comes in.

That said, the normal use case is: 1 user, 1 workspace. Or even if there are multiple users in a workspace, they are unlikely to be using it simultaneously. Whatever we come up with, that UX needs to be as simple as possible.

Reply:
IMHO, the only *good* UX for collaborative editing of documents is Google Drive style;  https://colab.research.google.com/ provides this. We probably can't do anything like that for now, unfortunately.

In the absence of that, researchers might just name their notebooks so they don't collide with each other; then there's no worries about clobbering each others' changes.

Reply:
(I have another idea for how to avoid having changes be accidentally clobbered... will sketch it up and send it out separate from this PR.)